libs/sdk-ui-web-components/src/parseUrl.ts

```diff
@@ -0,0 +1,40 @@
+// (C) 2022 GoodData Corporation
+
+import invariant from "ts-invariant";
+
+/**
+ * "none" means programmatic backend setup
+ * "sso" means automatic Tiger SSO with redirection
+ * "bearSso" means Service-provider initiated SAML SSO on Bear
+ * "bear" means no SSO, just redirect to /account.html on the Bear server
+ */
+export type AuthType = "none" | "sso" | "bearSso" | "bear";
+
+const mapAuthType = (auth: string | null): AuthType => {
+    switch (auth) {
+        case "sso":
+        case "bearSso":
+        case "bear":
+            return auth;
+        case "none":
+        default:
+            return "none";
+    }
+};
+
+/**
+ * @internal
+ *
+ * TODO rewrite this once specs on infra are clarified, versioned etc.
+ *  For now, assuming URL pattern: https://[host]/components/[workspaceId].js[?auth=sso]
+ */
+export const parseUrl = (scriptUrl: URL) => {
+    const protocol = scriptUrl.protocol.toLowerCase() === "http:" ? "http:" : "https:";
+    const host = scriptUrl.host;
+    const workspaceId = scriptUrl.pathname.match(/^\/components\/([^./]+)(\.js)?$/i)?.[1];
```

Comment:
Assuming workspaceId can't include dot (.) Not sure if this is really the case or I should make this RegExp more robust.

Reply:
On tiger and panther dots are allowed (just tested it), so please try to make this more robust.